src/selectors.cc

```diff
@@ -220,20 +220,31 @@ select_to_first_non_blank(const Context& context, const Selection& selection)
     return {it.coord()};
 }
 
+template<bool forward>
 Optional<Selection>
 select_matching(const Context& context, const Selection& selection)
 {
     auto& buffer = context.buffer();
     auto& matching_pairs = context.options()["matching_pairs"].get<Vector<Codepoint, MemoryDomain::Options>>();
     Utf8Iterator it{buffer.iterator_at(selection.cursor()), buffer};
     auto match = matching_pairs.end();
-    while (it != buffer.end())
+
+    if (forward) while (it != buffer.end())
     {
         match = find(matching_pairs, *it);
         if (match != matching_pairs.end())
             break;
         ++it;
     }
+    else while (true)
+    {
+        match = find(matching_pairs, *it);
+        if (match != matching_pairs.end()
+            or it == buffer.begin())
```

Comment:
I guess it doesn't matter either way, but why did you put the `it == buffer.begin()` inside the `if`, instead of in the `while` condition to match the other case?

Reply:
Because otherwise any opening delimiter at the very beginning of the buffer would not be considered for forward matching (the loop would quit as soon as the iterator reached the beginning of the buffer, without matching it with `matching_pairs`).